core/flow/tc_warm_up.go

```diff
@@ -0,0 +1,101 @@
+package flow
+
+import (
+	"math"
+	"sync/atomic"
+
+	"github.com/alibaba/sentinel-golang/core/base"
+	"github.com/alibaba/sentinel-golang/core/config"
+	"github.com/alibaba/sentinel-golang/util"
+)
+
+type WarmUpTrafficShapingCalculator struct {
+	threshold         float64
+	warmUpPeriodInSec uint32
+	coldFactor        uint32
+	warningToken      uint64
+	maxToken          uint64
+	slope             float64
+	storedTokens      *uint64
+	lastFilledTime    *uint64
+}
+
+func NewWarmUpTrafficShapingCalculator(rule *FlowRule) *WarmUpTrafficShapingCalculator {
+	if rule.WarmUpColdFactor <= 1 {
```

Comment:
May be we should check the WarmUpColdFactor value when load rules and give some warn logs just like  function checkControlBehaviorField.Otherwise users may feel different from what they expected

Reply:
Good suggestion, append print log reminder